### PR TITLE
ft: add versionId to multiObjDelete requestContext [S3C-184]

### PR DESCRIPTION
**Pull request by electrachong**
This is to add version ids to the request contexts for multi object delete requests.
Dependency of scality/Vault#728

### CODE REVIEW

File: lib/api/multiObjectDelete.js [hunk 1]
```diff
@@ -370,7 +370,10 @@ function multiObjectDelete(authInfo, request, log, callback) {
                     signatureAge: authParams.params.data.signatureAge,
                 },
                 parameterize: {
-                    specificResource: objects.map(entry => entry.key),
+                    specificResource: objects.map(entry => Object.assign({}, {
```
Comment (LaurenSpiegel) [inline lib/api/multiObjectDelete.js]: no need to do object.assign

**Comment by electrachong** [comment]
end to end passed here: http://ci.ironmann.io/gh/scality/Integration/11688

File: lib/api/multiObjectDelete.js [hunk 2]
```diff
@@ -412,11 +418,21 @@ function multiObjectDelete(authInfo, request, log, callback) {
...
                         }
+                        const key = result.arn.slice(slashIndex + 1);
+                        const versionId = result.versionId ?
+                            versionIdUtils.encode(result.versionId) : undefined;
                         if (result.isAllowed) {
-                            inPlay.push(objects[i]);
+                            inPlay.push({
+                                key,
+                                versionId,
+                                decodedVid: result.versionId,
```
Comment (LaurenSpiegel) [inline lib/api/multiObjectDelete.js]: add note as to why you save the decodedVid here

Comment (electrachong) [inline lib/api/multiObjectDelete.js]: I think this is wrong. We should be using the encoded version id sent by user for policies and so result.versionId should be the encoded version id. I need to look at this again

**Comment by electrachong** [comment]
end-to-end passed here: http://ci.ironmann.io/gh/scality/Integration/11765

**Pull request merged**